### ISSUE TITLE
scubakiz/servicedemo => changed listening port

### ISSUE THREAD

**Issue by lgmorand** [scubakiz/clusterinfo#6]
Hi

I delivering a workshop plus lab which is using your container. The lab contains manifest to deploy and expose your container using the port 80, but it's not working. By digging, I found your container is currently listening on port 8080.

Two possibility:
- the lab is wrong for years
- you changed the port on your side

When I look here, it seems that you pushed a new version a week ago by reusing an oldest tag: https://hub.docker.com/r/scubakiz/servicedemo/tags

By any chance, did you change the port ?

**Comment by ivanthelad** [comment]
Seem the image tag for serviceDemo:1.0 was updated recently., 9 days ago 
https://hub.docker.com/layers/scubakiz/servicedemo/1.0/images/sha256-186fc29ccf3d58e72a5e789e7b757f3ff9c7ca03b014f855f3a5f877fa1997bb?context=explore

**Comment by lgmorand** [comment]
yes, that's why I said "it seems that you pushed a new version a week ago by reusing an oldest tag" but I'd like a confirmation from the owner. I don't even understand how we can base a paid workshop on an external resource as a mistake can quickly happen. I'm not even sure this repo is still maintained by a MSFT employee :)

I could understand, why changing from 80 to 8080 to have a non root port but consequences are horrible here, especially reusing an oldest tag instead of having a v1.1

**Comment by scubakiz** [comment]
I just verified the Dockerfile and the container does listen to port 80.  Not sure where you're seeing 8080.  The service maps the port from 8100.

**Comment by scubakiz** [comment]
The repo is still maintained by me (a Microsoft employee).  If you DM me I will add you to the source code repo so you can make updates if needed (other than the port).

**Comment by scubakiz** [comment]
Just pushed 1.1 to avoid confusion.  No changes were made except it was compiled using .NET 8.0

**Comment by lgmorand** [comment]
I got the port 8080 by asking the logs of the pod

```PS C:\Users\lgmor> kubectl logs sample-dep-54766d879-jn689
warn: Microsoft.AspNetCore.DataProtection.Repositories.FileSystemXmlRepository[60]
info: Microsoft.Hosting.Lifetime[14]
      **Now listening on: http://[::]:8080**

      Content root path: /app
```

And as soon as we change the deployment manifest with 8080 instead of 80, it works. 


Furthermore, even testing locally shows that the container is not listening on port 80
```
docker run -d -p 999:80 --name test-scuba scubakiz/servicedemo:1.0 => broken
docker run -d -p 998:8080 --name test-scuba2 scubakiz/servicedemo:1.0 => OK
```

By upgrading to .Net core 8.0, you broke it inadvertently because of the behavior change: https://review.learn.microsoft.com/en-us/dotnet/core/compatibility/containers/8.0/aspnet-port?branch=main


I'm ok to help because the quick fix would be to regenerate a version 1.0 listening on port 80, either by changing the asp.net port, or by reverting to previous version of .net Core

**Comment by scubakiz** [comment]
This is the cause of the issue.  I updated the project to use .Net 8, but didn't realize the default port has changed:

https://learn.microsoft.com/en-us/dotnet/core/compatibility/containers/8.0/aspnet-port

I updated the project to explicitly use port 80.  You can get the working version by updating your image spec to:

scubakiz/servicedemo:1.2

The port is now set back to 80.
